ios/Classes/PdftronFlutterPlugin.m

```diff
@@ -2708,6 +2726,139 @@ - (void)openLayersList:(FlutterResult)flutterResult
+}
+
+-(void)openAddPagesView:(FlutterResult)flutterResult rect:(NSDictionary *)rect
+{
+    PTDocumentController *documentController = [self getDocumentController];
+    if(documentController == Nil)
+    {
+        // something is wrong, document view controller is not present
+        NSLog(@"Error: The document view controller is not initialized.");
+        flutterResult([FlutterError errorWithCode:@"open_add_pages_view" message:@"Failed to open add pages view" details:@"Error: The document view controller is not initialized."]);
+        return;
+    }
+    
+    NSNumber *rectX1 = rect[PTRectX1Key];
+    NSNumber *rectY1 = rect[PTRectY1Key];
+    NSNumber *rectX2 = rect[PTRectX2Key];
+    NSNumber *rectY2 = rect[PTRectY2Key];
+    CGRect screenRect = CGRectMake([rectX1 doubleValue], [rectY1 doubleValue], [rectX2 doubleValue]-[rectX1 doubleValue], [rectY2 doubleValue]-[rectY1 doubleValue]);
+    
+    [documentController showAddPagesViewFromScreenRect:screenRect];
+    flutterResult(nil);
+}
+
+-(void)openViewSettings:(FlutterResult)flutterResult rect:(NSDictionary *)rect
+{
+    PTDocumentController *documentController = [self getDocumentController];
+    if(documentController == Nil)
+    {
+        // something is wrong, document view controller is not present
+        NSLog(@"Error: The document view controller is not initialized.");
+        flutterResult([FlutterError errorWithCode:@"open_view_settings" message:@"Failed to open view settings" details:@"Error: The document view controller is not initialized."]);
+        return;
+    }
+    
+    NSNumber *rectX1 = rect[PTRectX1Key];
+    NSNumber *rectY1 = rect[PTRectY1Key];
+    NSNumber *rectX2 = rect[PTRectX2Key];
+    NSNumber *rectY2 = rect[PTRectY2Key];
+    CGRect screenRect = CGRectMake([rectX1 doubleValue], [rectY1 doubleValue], [rectX2 doubleValue]-[rectX1 doubleValue], [rectY2 doubleValue]-[rectY1 doubleValue]);
+    [documentController showSettingsFromScreenRect:screenRect];
+    flutterResult(nil);
+}
+
+-(void)openCrop:(FlutterResult)flutterResult
+{
+    // TODO: add a source rect option to the native API (similar to the openAddPagesView call)
+    PTDocumentController *documentController = [self getDocumentController];
+    if(documentController == Nil)
+    {
+        // something is wrong, document view controller is not present
+        NSLog(@"Error: The document view controller is not initialized.");
+        flutterResult([FlutterError errorWithCode:@"open_crop" message:@"Failed to open crop" details:@"Error: The document view controller is not initialized."]);
+        return;
+    }
+    PTFlutterDocumentController *flutterDocumentController = (PTFlutterDocumentController *) documentController;
+    if (flutterDocumentController.isTopToolbarsHidden) {
+        return;
+    };
+    if (documentController.navigationItem.rightBarButtonItem != nil) {
+       [documentController showPageCropOptions:documentController.navigationItem.rightBarButtonItem];
+    }
```

Comment:
This is a temporary workaround at my request to prevent a crash on iPads.
The long term solution is to ensure that the `UIAlertController` shown from the `showPageCropOptions:` method can have a `sourceRect` specified along with a `sourceView` in the case where there is no `UIBarButtonItem` to use as the popover anchor.